crates/ruff_server/CONTRIBUTING.md

```diff
@@ -0,0 +1,32 @@
+## Contributing to the Ruff Language Server
+
+This is a mostly free-form guide with resources to help you get started with contributing to `ruff server`.
+
+### Project Architecture
+
+`ruff_server` uses a lock-free data model to represent its state. The server runs in a continuous 'event loop' by listening to incoming messages
```

Comment:
Is there a file we could link to from `data model`. Like is there a datastructure that stores the entire editor state? 

I think it would also be helpful to add other links. E.g link `event loop`  to the event loop in the implementation. Link `local` or `background` to the corresponding definitions. Another useful link could be around *Snapshots of the server*.